### PR TITLE
Add HDFS OPA authorizer

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/hdfs-operator/issues/400

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] (Integration-)Test cases added
- [x] Documentation added or updated
- [ ] Changelog updated
```

```[tasklist]
# Acceptance
- [ ] Proper release label has been added
```

### CODE REVIEW

File: test/stack/11-rego-rules.yaml
Comment (adwk67) [inline test/stack/11-rego-rules.yaml:26]: I haven't tested it but this looks like it will return a result without the "groups" subnesting shown in the comment directly above these lines. Are we sure that the output is the same? (I tried to reproduce my sample data based on the UIF output from another example elsewhere, with named fields).

Comment (sbernauer) [reply test/stack/11-rego-rules.yaml:26]: I'm not sure to be honest. I can only say the current PR works flawlessly (both the autorizer and the group mapper)
The response might also depend on whether you call http://opa.default.svc.cluster.local:8081/v1/data/hdfs/groups or http://opa.default.svc.cluster.local:8081/v1/data/hdfs

Comment (adwk67) [reply test/stack/11-rego-rules.yaml:26]: Yes, looking at the code I see the rule output matches what the mapper expects. And should the UIF output change, we can always take of that in the rules (much like we do with trimming the trailing slash).

File: README.md
Comment (adwk67) [inline README.md]: I don't think this is correct as the mapper now expects a list and not a map.

Comment (sbernauer) [reply README.md]: c1aaaf6

File: test/stack/11-rego-rules.yaml
Comment (sbernauer) [reply test/stack/11-rego-rules.yaml:26]: c1aaaf6

Comment (adwk67) [reply test/stack/11-rego-rules.yaml:26]: Thanks!

**Review by adwk67 — APPROVED** [review]
LGTM! As discussed, I haven't looked at the rego rules in depth as they will be tested elsewhere anyway. The mapping stuff looks good. Approving so it doesn't block further work.

**Comment by sbernauer** [comment]
Just for reference, the Issue to create "production-ready" rego rules is https://github.com/stackabletech/issues/issues/523

File: test/stack/20-hdfs.yaml
Comment (NickLarsenNZ) [inline test/stack/20-hdfs.yaml]: optional, but as discussed we can do this to remove the false-positives:

```suggestion
      tlsSecretClass: tls # pragma: allowlist secret
```

File: test/stack/20-hdfs.yaml
Comment (NickLarsenNZ) [inline test/stack/20-hdfs.yaml]:
```suggestion
        secretClass: kerberos-default # pragma: allowlist secret
```

File: test/stack/30-test-hdfs-permissions.yaml
Comment (NickLarsenNZ) [inline test/stack/30-test-hdfs-permissions.yaml:2]:
```suggestion
---
apiVersion: batch/v1
```

**Review by NickLarsenNZ — APPROVED** [review]
Just added suggestions for the things the linter picked up.

Docs: https://github.com/reviewdog/action-detect-secrets?tab=readme-ov-file#false-positives (Thanks @Techassi)

File: src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java
Comment (adwk67) [inline src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java:125]: Nit: maybe log out the non-pretty-printed query here before the throw, as it may be lost in the stacktrace (or even harder to find!)

**Review by adwk67 — APPROVED** [review]
lgtm. Left a comment about logging the non-pp query, but I leave that up to you.

File: src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java
Comment (sbernauer) [reply src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java:125]: 7e0ef4bb12e1bdf50a2780a614748b964d4d848e

**Review by adwk67 — APPROVED** [review]

File: src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java
Comment (adwk67) [reply src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java:125]: Thanks!

**Pull request merged**
